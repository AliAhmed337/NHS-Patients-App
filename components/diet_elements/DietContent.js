import React from 'react';
import { Ionicons } from '@expo/vector-icons';
import { View, Text } from 'react-native';

export const DietContent = (props) => {

	let allowedString = props.isAllowed.reduce((acc, o) => acc ? `${acc}, ${o.name}` : o.name, null);
	let notAllowedString = props.notAllowed.reduce((acc, o) => acc ? `${acc}, ${o.name}` : o.name, null);

	return (
		<View>
		<View style={{ flexWrap: 'wrap', flexDirection: 'row' }}>
			<View style={{ flexBasis: '50%' }}>
				<View style={{ flexDirection: 'row', margin: 5 }}>
				<Text style={styles.titleStyle}>Allowed:</Text>
				</View>
				<Text style={styles.contentStyle}>{allowedString ? allowedString : "None"}</Text>
			</View>
			<View style={{ flexBasis: '50%' }}>
				<View style={{ flexDirection: 'row', margin: 5 }}>
				<Text style={styles.titleStyle}>Not Allowed:</Text>
				</View>
				<Text style={styles.contentStyle}>{notAllowedString ? notAllowedString : "None"}</Text>
			</View>
		</View>
		</View>
	);
};

const styles = {
	titleStyle: {
		color: '#003087',
		fontWeight: 'bold',
		textDecorationColor: '#003087',
		fontSize: 19,
		margin: 5,
	},
	contentStyle: {
		color: 'black',
		fontSize: 16, 
		margin: 10, 
	}
};