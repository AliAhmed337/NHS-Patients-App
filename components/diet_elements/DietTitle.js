import React from 'react';
import { MaterialCommunityIcons } from '@expo/vector-icons';
import { View, Text } from 'react-native';

export const DietTitle = (props) => {
	return (
		<View style={styles.dietTitleArrangementStyle}>
		    <MaterialCommunityIcons name="chevron-down-circle-outline" size={25} color="blue" />
			<Text style={styles.dietTitleTextStyle}>{props.title}</Text>
		</View>
	);
};

const styles = {
	dietTitleTextStyle: {
		fontSize: 20, 
		fontWeight: '400',
		color: '#005EB8', 
	},
	dietTitleArrangementStyle: {
		flexDirection: 'row',
		justifyContent: 'space-around'
	}
}