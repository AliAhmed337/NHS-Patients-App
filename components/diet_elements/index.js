export * from './DietTitle';
export * from './DietContent';