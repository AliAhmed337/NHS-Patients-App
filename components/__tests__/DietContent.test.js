import React from 'react';
import { shallow } from 'enzyme';
import { DietContent } from '../diet_elements';

describe('Testing diet content component', () => {

	it('renders correcly', () => {
		const wrapper = shallow(
			<DietContent allowed="allowed" notAllowed="notAllowed" />
			);
		wrapper.setProps({ allowed: 'allowed', notAllowed: 'notAllowed' });
		expect(wrapper).toMatchSnapshot();

	});	

});
