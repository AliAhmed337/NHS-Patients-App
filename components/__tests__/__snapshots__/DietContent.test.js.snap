// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing diet content component renders correcly 1`] = `
<Component>
  <Component
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
      }
    }
  >
    <Component
      style={
        Object {
          "flexBasis": "50%",
        }
      }
    >
      <Component
        style={
          Object {
            "flexDirection": "row",
            "margin": 5,
          }
        }
      >
        <Component
          style={
            Object {
              "color": "#003087",
              "fontSize": 19,
              "fontWeight": "bold",
              "margin": 5,
              "textDecorationColor": "#003087",
            }
          }
        >
          Allowed:
        </Component>
      </Component>
      <Component
        style={
          Object {
            "color": "black",
            "fontSize": 16,
            "margin": 10,
          }
        }
      >
        allowed
      </Component>
    </Component>
    <Component
      style={
        Object {
          "flexBasis": "50%",
        }
      }
    >
      <Component
        style={
          Object {
            "flexDirection": "row",
            "margin": 5,
          }
        }
      >
        <Component
          style={
            Object {
              "color": "#003087",
              "fontSize": 19,
              "fontWeight": "bold",
              "margin": 5,
              "textDecorationColor": "#003087",
            }
          }
        >
          Not Allowed:
        </Component>
      </Component>
      <Component
        style={
          Object {
            "color": "black",
            "fontSize": 16,
            "margin": 10,
          }
        }
      >
        notAllowed
      </Component>
    </Component>
  </Component>
</Component>
`;
